cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -52,212 +49,333 @@ func TestConstructor(t *testing.T) {
 	assert.NotNil(t, newConsumer)
 }
 
-// partitionConsumerWrapper wraps a Sarama partition consumer into a Sarama cluster partition consumer
-type partitionConsumerWrapper struct {
-	topic     string
-	partition int32
-
-	sarama.PartitionConsumer
+func TestSaramaConsumerWrapper_MarkPartitionOffset(t *testing.T) {
+	sc := &kmocks.Consumer{}
+	metadata := "meatbag"
+	sc.On("MarkPartitionOffset", topic, partition, msgOffset, metadata).Return()
+	sc.MarkPartitionOffset(topic, partition, msgOffset, metadata)
+	sc.AssertCalled(t, "MarkPartitionOffset", topic, partition, msgOffset, metadata)
 }
 
-func (s partitionConsumerWrapper) Partition() int32 {
-	return s.partition
+// NewTestConfig returns a config meant to be used by tests.
+// Due to inconsistencies with the request versions the clients send using the default Kafka version
+// and the response versions our mocks use, we default to the minimum Kafka version in most tests
+func NewTestConfig() *sarama.Config {
+	config := sarama.NewConfig()
+	config.Version = sarama.MinVersion
+	return config
 }
 
-func (s partitionConsumerWrapper) Topic() string {
-	return s.topic
+type Store struct {
+	store     *memory.Store
+	consumer  *Consumer
+	done      sync.WaitGroup
+	t         *testing.T
+	returnErr bool
 }
 
-func newSaramaClusterConsumer(saramaPartitionConsumer sarama.PartitionConsumer, mc *smocks.PartitionConsumer) *kmocks.Consumer {
-	pcha := make(chan cluster.PartitionConsumer, 1)
-	pcha <- &partitionConsumerWrapper{
-		topic:             topic,
-		partition:         partition,
-		PartitionConsumer: saramaPartitionConsumer,
+func (s *Store) WriteSpan(ctx context.Context, span *model.Span) error {
+	err := s.store.WriteSpan(ctx, span)
+	s.t.Log("Write span store!!!")
+	go func() {
+		s.consumer.Close()
+		s.done.Done()
+	}()
+	if err == nil && s.returnErr {
+		err = errors.New("Write span store failed")
 	}
-	saramaClusterConsumer := &kmocks.Consumer{}
-	saramaClusterConsumer.On("Partitions").Return((<-chan cluster.PartitionConsumer)(pcha))
-	saramaClusterConsumer.On("Close").Return(nil).Run(func(args mock.Arguments) {
-		mc.Close()
-	})
-	saramaClusterConsumer.On("MarkPartitionOffset", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
-	return saramaClusterConsumer
+	return err
 }
 
-func newConsumer(
-	t *testing.T,
-	metricsFactory metrics.Factory,
-	topic string,
-	processor processor.SpanProcessor,
-	consumer consumer.Consumer,
-) *Consumer {
-	logger, _ := zap.NewDevelopment()
-	consumerParams := Params{
-		MetricsFactory:   metricsFactory,
-		Logger:           logger,
-		InternalConsumer: consumer,
-		ProcessorFactory: ProcessorFactory{
-			topic:          topic,
-			consumer:       consumer,
-			metricsFactory: metricsFactory,
-			logger:         logger,
-			baseProcessor:  processor,
-			parallelism:    1,
+func TestConsumerGroupNewSessionDuringOffsetLoad(t *testing.T) {
```

Comment:
I could not find an `assert` or `require` package call in this test as well as `TestConsumerGroupForReturnError`. 

What are you trying to test here and in `TestConsumerGroupForReturnError`?

Reply:
test case `TestConsumerGroupForReturnError ` is invalid, removed it,`require` is already used in consumer_test.go files to handle errors